### PR TITLE
add open_in option to coder_app

### PR DESCRIPTION
**Pull request by defelmnq**
Working on #297

### CODE REVIEW

**Review by mtojek — COMMENTED** [review]
Linter complains about docs, AFAIR `make gen` command should help here?

**Review by mafredri — APPROVED** [review]
Just a few thoughts on docs, otherwise LGTM 👍🏻

File: provider/app.go [hunk 1]
```diff
@@ -223,6 +223,28 @@ func appResource() *schema.Resource {
 				ForceNew:    true,
 				Optional:    true,
 			},
+			"open_in": {
+				Type: schema.TypeString,
+				Description: "Determines where the app will be opened. Valid values are `\"tab\"`, `\"window\"`, and `\"slim-window\" (default)`. " +
+					"`\"tab\"` opens in a new tab in the same browser window. `\"window\"` opens a fresh browser window with navigation options. " +
+					"`\"slim-window\"` opens a fresh browser window with slim navigation options.",
```
[the remark this answers is not in the document]
Comment (mafredri) [reply provider/app.go:230]: On second thought, toolbar is probably superfluous so I'll strike it from the suggestion.

**Review by johnstcn — APPROVED** [review]

**Comment by matifali** [comment]
Will it work the same for `external=true`? 
I am concerned about VSCode-Desktop, Cursor, and JetBrains Gateway modules. Please make sure that we do not leave an empty window when a user clicks on a `coder_app` that opens a URI that launches another app.

**Comment by defelmnq** [comment]
> Will it work the same for `external=true`? I am concerned about VSCode-Desktop, Cursor, and JetBrains Gateway modules. We should make sure that we do not leave an empty window when a user clicks on a `coder_app` that opens a URI that launches another app.

Yes, so for now it will - for the apps not-browser based - just be ignored. Anyway, your point makes complete sense and it can be a follow-up item to have validation to either have either external or open_in parameters.

**Review by mtojek — APPROVED** [review]

**Pull request merged**